src/shared.c

```diff
@@ -814,6 +822,50 @@ _Bool GlobMatch(const char *pattern, const char *text) {
 }
 
 /**
+ * @brief Matches the pattern against specified text, returning true if the pattern
+ * matches, false otherwise.
+ *
+ * A match means the entire string TEXT is used up in matching.
+ *
+ * In the pattern string, `*' matches any sequence of characters,
+ * `?' matches any character, [SET] matches any character in the specified set,
+ * [!SET] matches any character not in the specified set.
+ *
+ * A set is composed of characters or ranges; a range looks like
+ * character hyphen character(as in 0-9 or A-Z).
+ * [0-9a-zA-Z_] is the set of characters allowed in C identifiers.
+ * Any other character in the pattern must be matched exactly.
+ *
+ * To suppress the special syntactic significance of any of `[]*?!-\',
+ * and match the character exactly, precede it with a `\'.
+ */
+_Bool GlobMatch(const char *pattern, const char *text) {
+	return GlobMatch_(pattern, text, true);
+}
+
+/**
+ * @brief Matches the pattern against specified text case-insensitively, returning true
+ * if the pattern matches, false otherwise.
+ *
+ * A match means the entire string TEXT is used up in matching.
+ *
+ * In the pattern string, `*' matches any sequence of characters,
+ * `?' matches any character, [SET] matches any character in the specified set,
+ * [!SET] matches any character not in the specified set.
+ *
+ * A set is composed of characters or ranges; a range looks like
+ * character hyphen character(as in 0-9 or A-Z).
+ * [0-9a-zA-Z_] is the set of characters allowed in C identifiers.
+ * Any other character in the pattern must be matched exactly.
+ *
+ * To suppress the special syntactic significance of any of `[]*?!-\',
+ * and match the character exactly, precede it with a `\'.
+ */
+_Bool GlobiMatch(const char *pattern, const char *text) {
+	return GlobMatch_(pattern, text, false);
+}
```

Comment:
My main critique of these functions (`Cmd_Get_`, `Cvar_Get_`, and `GlobMatch_`) is that taking in a _Bool in c has become a bit of a smell for me. I'm actually in the process of refactoring `Fs_Init` for this same smell, because I now have an additional flag to pass in.

Might it be better to define a bitmask of flags, and take in `uint32_t flags` instead? You could then `#define GLOB_CASE_INSENSITIVE` and potentially (in the future) some others (`GLOB_REGEX`?). What do you think?